### DISCUSSION TITLE
Usage without TypeScript?

### DISCUSSION THREAD

**Discussion by thevipinmishra** [vanilla-extract-css/vanilla-extract#817] [Q&A]
Can this be used without TypeScript?

I mean my project is on Next.js and we are not using Typescript or any type-checking in our code!

Do we need typescript for this? and if yes,  how to set it up ?

**Comment by mattcompiles** [comment]
You don't need to use TypeScript to use vanilla-extract. You'll just miss out on all the type safety it provides. 

If you'd like to try TypeScript, I'd suggest following the next documentation on how to setup TypeScript. Nothing special is required from a vanilla-extract perspective.
